agent/src/softwarecontainerfactory.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+/**
+ * @file softwarecontainerfactory.h
+ * @brief Contains the softwarecontainer::AgentFactory class
+ */
+#pragma once
+
+#include "softwarecontainer.h"
+#include "config/softwarecontainerconfig.h"
+#include "softwarecontainer-common.h"
+
+namespace softwarecontainer {
+
+/*
+ * This class holds a connection to agent to create containers.
```

Comment:
Not it doesn't, it is a class that creates containers. Its relation to the agent should be described in the agent, not here.